src/Relogify/AddResult.fs

```diff
@@ -1,48 +1,125 @@
 module Relogify.AddResult
 
+open System
 open Fabulous
 open Fabulous.XamarinForms
 open Xamarin.Forms
+open Relogify.Graphql
 
-type Model =
-    { OwnName: string
-      OpponentName: string
-      OwnPoints: int
+type ResultModel =
+    { OwnPoints: int
       OpponentPoints: int
-      ExtraTime: bool
-      CanAddResult: bool }
+      ExtraTime: bool }
+
+type State =
+    | EditingResult
+    | AddingResult
+    | ErrorAddingResult of string
+
+type Model =
+    { resultModel: ResultModel
+      state: State }
 
 let initModel =
-    { OwnName = "Player1"
-      OpponentName = "Player2"
-      OwnPoints = 0
-      OpponentPoints = 0
-      CanAddResult = false
-      ExtraTime = false }
+    { resultModel =
+        { OwnPoints = 0
+          OpponentPoints = 0
+          ExtraTime = false }
+      state = EditingResult }
 
 type Msg =
     | SetOwnPoints of int
     | SetOpponentPoints of int
     | ToggleExtraTime
+    | AddResultInitiated
+    | ResultAddedSuccess
+    | ResultAddedError of string
+
+type AddResultModel =
+    { CommunityName: string
+      PlayerName: string
+      OpponentName: string
+      PlayerPoints: int
+      OpponentPoints: int
+      ExtraTime: bool }
+
+type CmdMsg =
+    | Noop
+    | AddResultCmdMsg of AddResultModel
+
+let rand = Random()
 
-type CmdMsg = Noop
+let addResultCmd (resultModel: AddResultModel) =
+    async {
+        // replace getPlayersOperation with addResultOperation
+        do! Async.Sleep 1000
+        let! result = getPlayersOperation.AsyncRun(runtimeContext, resultModel.OpponentName) // TODO: Send in the real data
+        return
+            match result.Data with // TODO: result.Errors
+            | Some _ -> // ResultAddedSuccess // TODO: Uncomment this and remove the randomness
+                if rand.Next(2) = 0 then
+                    ResultAddedError "Failed to add result, please check your internet connection and try again"
+                else
+                    ResultAddedSuccess
+            | None -> ResultAddedError "Failed to add result, please check your internet connection and try again"
+    }
+    |> Cmd.ofAsyncMsg
 
 let mapCommands: CmdMsg -> Cmd<Msg> =
     function
     | Noop -> Cmd.none
+    | AddResultCmdMsg resultModel -> addResultCmd resultModel
 
-let update model msg: Model * CmdMsg list =
+let toAddResultModel (playerName: string) (opponentName: string) (communityName: string) (resultModel: ResultModel): AddResultModel =
```

Comment:
Maybe remove most of the type annotations? Looks a bit "tjatigt" :)

Reply:
Removed the strings, but keeping the rest, since it might help in case the inference gets in the way (might easily happen with `AddResultModel` and `ResultModel`).